src/Engine/GameEvents/MapChangeEvent.java

```diff
@@ -45,30 +38,56 @@ public GameAnimation getEventAnimation(MapView mapView)
   @Override
   public void sendToListener(GameEventListener listener)
   {
-    listener.receiveMapChangeEvent(this);
+    listener.receiveTerrainChangeEvent(changes);
   }
 
   @Override
   public void performEvent(MapMaster gameMap)
   {
-    Location loc = gameMap.getLocation(where);
-    if( null != loc )
+    for( EnvironmentAssignment ea : changes )
     {
-      loc.setEnvironment(myEnvironment);
-      if (turns > 0)
-        loc.setForecast(myEnvironment.weatherType, (gameMap.commanders.length * turns) - 1);
```

Comment:
This math is needed because the basic unit of duration is a turn rather than a full round.
1 is subtracted in order to ensure the weather reverts on the same CO's turn as it started.